### DISCUSSION TITLE
Code Review & Refactoring

### DISCUSSION THREAD

**Discussion by bastoune** [PhilippeMarcMeyer/vanillaSelectBox#86] [Ideas]
Hi,

I am working with this great tool in a bunch of projects.

I did some refactoring and want to do a lot more.

I would like to know if I can submit suggestions with a "big" PR. And for that I would like to know if you want to keep compatibility with old navigators ?
Is there a minimum version of ECMAScript would you like the code to be compatible with ? Which one ?

Would it bother you if I split the code in multiple "functionnal" files ?

Would you agree on adding some Unit Testing with Jest ?

**Comment by PhilippeMarcMeyer** [comment]
Sorry I didn't check your mail sooner !
I don't bother being compatible with oldies, let's put them behind us !
ES6 is great and widly supported by now
One thing I don't want is to complexify the plugin
We could make another version supporting n levels but not in vanillaSelectBox please : let's keep it simple with a real SELECT tag as a source of data

Sincerely

Philippe
